### PR TITLE
[openblas] Fix build errors on osx

### LINKED ISSUE #36031: [openblas] Build error on arm64-android
It's failing with ```ld: error: undefined symbol: cabsf```

I think the complex.h header is missing from a file.

 [issue_body.md](https://github.com/microsoft/vcpkg/files/13832246/issue_body.md)

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/36031

```
ld: error: undefined symbol: cabsf
>>> referenced by c_cblat2c.c:3566 (/home/CORP/b.manganelli/Projects/vcpkg/buildtrees/openblas/src/v0.3.25-dc6d649124.clean/ctest/c_cblat2c.c:3566)
>>>               ctest/CMakeFiles/xccblat2.dir/c_cblat2c.c.o:(cmvch_)
```

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

Usage test pass with following triplets:

```
x64-windows
x64-osx
```

### CODE REVIEW

File: ports/openblas/portfile.cmake [hunk 1]
```diff
@@ -32,7 +32,9 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 
 set(COMMON_OPTIONS -DBUILD_WITHOUT_LAPACK=ON)
 
+set(BLAS_OPTIONS "")
```
Comment (dg0yt) [inline ports/openblas/portfile.cmake:35]: This is the fourth options variable... Can't it be integrated?

**Review by dg0yt — COMMENTED** [review]
Unrelated to this PR: The port doesn't correctly acquire git (if needed at all) and sed.

File: ports/openblas/portfile.cmake [hunk 2]
```diff
@@ -33,6 +33,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 set(COMMON_OPTIONS -DBUILD_WITHOUT_LAPACK=ON)
 
 if(VCPKG_TARGET_IS_OSX)
+    list(APPEND COMMON_OPTIONS -DONLY_CBLAS=1)
```
Comment (bmanga) [inline ports/openblas/portfile.cmake:36]: Does this change also make it work for android?

Comment (jimwang118) [reply ports/openblas/portfile.cmake:36]: Added option separately for Android.

Comment (dg0yt) [reply ports/openblas/portfile.cmake:36]: > Added option separately for Android.

Did you test it? Android is skipped in ci.baseline.

Comment (jimwang118) [reply ports/openblas/portfile.cmake:36]: @dg0yt When I test on local Android, the following error occurs. My local environment is an Android environment built under Linux.
`When cross compiling, a TARGET is required.`
@bmanga Can you help with testing on your Android environment?

Comment (dg0yt) [reply ports/openblas/portfile.cmake:36]: So you were adding untested changes...
I prefer to start with logs from vcpkg CI before spending local build time.

Comment (jimwang118) [reply ports/openblas/portfile.cmake:36]: Judging from the upstream [code](https://github.com/OpenMathLib/OpenBLAS/blob/d6a5174e9c50b9f68db96d7d7818f92cdfb4e7ec/cmake/system.cmake#L10-L19), only arm64 and arm architectures are supported when compiling Android, and others are not supported. So my local test fails.

Comment (dg0yt) [reply ports/openblas/portfile.cmake:36]: The conclusion is wrong. Only arm64 and arm are *detected*. For others, you are asked to define `TARGET`.
And I only ask that the android shall be tested before merged. If possible, by removing the ci.baseline entries.

Comment (dg0yt) [reply ports/openblas/portfile.cmake:36]: Possible targets are found here:
https://github.com/OpenMathLib/OpenBLAS/blob/d6a5174e9c50b9f68db96d7d7818f92cdfb4e7ec/TargetList.txt
The (Android x64) task is to make a reasonable choice in the portfile. And the user may even override in the triplet file.

Comment (jimwang118) [reply ports/openblas/portfile.cmake:36]: I queried the CPU model of the device, but it was not in the support list provided by xxx and therefore could not be tested.
`16  AMD EPYC 7452 32-Core Processor`

Comment (bmanga) [reply ports/openblas/portfile.cmake:36]: @jimwang118 this PR works for me, thanks

Comment (jimwang118) [reply ports/openblas/portfile.cmake:36]: Thanks for testing.

**Comment by bmanga** [comment]
Any updates on this?

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**